src/dbnode/encoding/ostream.go

```diff
@@ -73,22 +87,50 @@ func (os *ostream) hasUnusedBits() bool {
 
 // grow appends the last byte of v to rawBuffer and sets pos to np.
 func (os *ostream) grow(v byte, np int) {
+	os.ensureCapacityFor(1)
+	os.rawBuffer = append(os.rawBuffer, v)
+
+	os.pos = np
+}
+
+// ensureCapacity ensures that there is at least capacity for n more bytes.
+func (os *ostream) ensureCapacityFor(n int) {
+	var (
+		currCap      = cap(os.rawBuffer)
+		currLen      = len(os.rawBuffer)
+		availableCap = currCap - currLen
```

Comment:
Nit: Given that availableCap is only used in the next line, may as well just do the calc of missingCap in 1 line?